### PR TITLE
[sycl-free-function] migrate UnfoldBackward kernels to sycl free function (#152)

### PR DESCRIPTION
**Pull request by jiafuzha**
Test results (all passing):                                                                                                                            
                                                                                                                                                            
      -  test_unfold_backward_errors_cpu  — OK                                                                                                              
      -  test_unfold_backward_errors_xpu  — OK                                                                                                              
      -  test_ops_gradients_xpu.py -k unfold  — 30 tests, 12 ran OK, 18 skipped (expected: no inplace variant / no gradgrad support for those variants) —  covers  unfold ,  unfold_copy ,  nn.functional.unfold  for float64/complex128, exercising the migrated free-function kernel via  gradcheck/gradgradcheck.                                                                                     
      - Manual sanity check:  x.unfold(...).sum().backward()  on XPU matches CPU gradient exactly.

### CODE REVIEW

**Comment by jiafuzha** [comment]
@torchxpubot ut-check

**Comment by jiafuzha** [comment]
@torchxpubot ut-check

**Review by gujinghui — APPROVED** [review]

**Comment by jiafuzha** [comment]
@majing921201 please help review again.

**Review by majing921201 — APPROVED** [review]

**Review by draghan — APPROVED** [review]

**Comment by tye1** [comment]
@torchxpubot  check-ut
